app.R

```diff
@@ -150,180 +150,159 @@ shinyApp (
     updateSelectizeInput(session, 'single_gene_name_3_t', choices = genenames_choices$Gene, server = TRUE)
     updateSelectizeInput(session, 'single_gene_name_4_t', choices = genenames_choices$Gene, server = TRUE)
     updateSelectizeInput(session, 'single_gene_name_5_t', choices = genenames_choices$Gene, server = TRUE)
-    
-    
-    
-    
+
+
+
+
     # MULTIPLOT #
     output$multigene_plot_tmt <- renderPlotly({
       # Getting actual genes to plot
-      #genesToPlot <- input$single_gene_name_1_t
-      genesToPlot <- c("DNMT3A","TP53")
+      genesToPlot <- input$single_gene_name_1_t
+      #genesToPlot <- c("DNMT3A","TP53")
       if(length(genesToPlot) > 0)
       {
-        tcga = NULL
+        # TODO: Make db a Variable
+        #SELECT UPN,Gene,Value FROM Genes WHERE (Gene='DNMT3A' OR Gene='TP53') AND Type='TMT';
+        query <- "SELECT UPN,Gene,Value FROM Genes WHERE ("
         for(g in genesToPlot){
-          tcga = rbind(tcga,read.table(paste0("/home/payton/research/washu/TCGA_Proteomics_App/Prep/TCGA_Proteomics_app/Gene_files/TMT/",g,".tsv"),sep="\t",stringsAsFactors=F,header=T,check.names=FALSE))
-          #tcga = rbind(tcga,read.table(paste0("https://storage.googleapis.com/tcga_shiny/TCGA_Proteomics_app/Gene_files/TMT/",g,".tsv"),sep="\t",stringsAsFactors=F,header=T,check.names=FALSE))
-          #SELECT Gene,UPN,Value FROM Genes WHERE (Gene='DNMT3A' OR Gene='TP53') AND Type='TMT';
+          query <- paste(query,sep="","Gene='",g,"'"," OR ")
         }
-        
-        # Data cleanup and preparing for output
-        df = melt(tcga,id.vars="gene")
-        names(df) <- c("Gene", "UPN", "Log2.Expression")
-        df <- merge(df, clinicaldata_tmt, by="UPN") # SELECT * FROM TMT_Clinical WHERE (UPN="ND8" OR ...);
+        query <- paste(sep="",substring(query,1, nchar(query)-4),")")
```

Comment:
I think this for loop and subsequent cleanup could be replaced with something like this to collapse the gene names into a string:

`paste("Gene='",genes,"'",sep="",collapse=" OR ")`

I think there are a few other places below where this pattern could be applied as well